site/app/libraries/database/DatabaseQueries.php

```diff
@@ -6063,10 +6063,10 @@ public function getQueueDataStudent() {
                 user_id,
                 COUNT(*) AS queue_interactions,
                 COUNT(DISTINCT name) AS number_names_used,
-                AVG(time_out - time_help_start) AS avg_help_time,
+                DATE_TRUNC('second', AVG(time_out - time_help_start)) AS avg_help_time,
                 MIN(time_out - time_help_start) AS min_help_time,
                 MAX(time_out - time_help_start) AS max_help_time,
-                AVG(time_help_start - time_in) AS avg_wait_time,
+                DATE_TRUNC('second', AVG(time_help_start - time_in)) AS avg_wait_time,
                 MIN(time_help_start - time_in) AS min_wait_time,
                 MAX(time_help_start - time_in) AS max_wait_time,
                 SUM(CASE
```

Comment:
Can you extract the common part of all of these queries (it looks like would cover it:

```
                 COUNT(*) AS queue_interactions,
                 COUNT(DISTINCT name) AS number_names_used,
                 COUNT(DISTINCT user_id) AS number_distinct_students,
                 AVG(time_out - time_help_start) AS avg_help_time,
                 DATE_TRUNC('second', AVG(time_out - time_help_start)) AS avg_help_time,
                 MIN(time_out - time_help_start) AS min_help_time,
                 MAX(time_out - time_help_start) AS max_help_time,
                 AVG(time_help_start - time_in) AS avg_wait_time,
                 DATE_TRUNC('second', AVG(time_help_start - time_in)) AS avg_wait_time,
                 MIN(time_help_start - time_in) AS min_wait_time,
                 MAX(time_help_start - time_in) AS max_wait_time,
                 SUM(CASE
                   WHEN removal_type IN ('helped', 'self_helped') THEN 1
                   ELSE 0
                 END) AS help_count,
                 SUM(CASE
                   WHEN removal_type IN ('removed', 'emptied', 'self') THEN 1
                   ELSE 0
                 END) AS not_helped_count
```
)

into a private function that then gets called and appended into these queries?

e.g.

```
private function getInnerQueueSelect(): string {
    return <<<SQL
                 COUNT(*) AS queue_interactions,
                 COUNT(DISTINCT name) AS number_names_used,
                 COUNT(DISTINCT user_id) AS number_distinct_students,
                 AVG(time_out - time_help_start) AS avg_help_time,
                 DATE_TRUNC('second', AVG(time_out - time_help_start)) AS avg_help_time,
                 MIN(time_out - time_help_start) AS min_help_time,
                 MAX(time_out - time_help_start) AS max_help_time,
                 AVG(time_help_start - time_in) AS avg_wait_time,
                 DATE_TRUNC('second', AVG(time_help_start - time_in)) AS avg_wait_time,
                 MIN(time_help_start - time_in) AS min_wait_time,
                 MAX(time_help_start - time_in) AS max_wait_time,
                 SUM(CASE
                   WHEN removal_type IN ('helped', 'self_helped') THEN 1
                   ELSE 0
                 END) AS help_count,
                 SUM(CASE
                   WHEN removal_type IN ('removed', 'emptied', 'self') THEN 1
                   ELSE 0
                 END) AS not_helped_count
SQL;
}
```

We should try and follow the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle here, and make it easier to modify the next time through.